django/forms/widgets.py

```diff
@@ -387,6 +387,9 @@ def value_from_datadict(self, data, files, name):
     def value_omitted_from_data(self, data, files, name):
         return name not in files
 
+    def use_required_attribute(self, initial):
+        return super().use_required_attribute(initial) and not initial
```

Comment:
We can remove `ClearableFileInput.use_required_attribute()` after this change.

Reply:
Okay. So should I remove `use_required_attribute()` from `ClearableFileInput` after including tests for `FileInput.use_required_attribute()` ? And should I make changes to both `FileInput` as well as `ClearableFileInput` documentation mentioning the behavior of  `user_required_attribute()` in the respective widget ?

Reply:
This PR doesn't change anything in the `ClearableFileInput` behavior and it is already well documented.

Reply:
Okay. I will then add the test and document the behavior for `FileInput.use_required_attribute()` in docs